### PR TITLE
[windows] Add ability to tag IIS services by site name in addition to…

### PR DESCRIPTION
**Pull request by derekwbrown**
… site id.

### Describe how to test/QA your changes
Install windows with IIS.
install agent
Generate traffic to IIS.
hit the debugging endpoint:
`iwr -UseBasicParsing -DisableKeepAlive http://localhost:3333/network_tracer/debug/http_monitoring | convertfrom-json`

You should see output indicating that the site id and site name are now correct, e.g.

`DynamicTags : {http.iis.app_pool:DefaultAppPool, http.iis.site:1, http.iis.sitename:Default Web Site,
              service:DefaultAppPool}`

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/etw/etw-http-service.go [hunk 1]
```diff
@@ -746,8 +751,9 @@ func httpCallbackOnHTTPRequestTraceTaskDeliver(eventInfo *C.DD_ETW_EVENT_INFO) {
 		return
 	}
 
-	httpConnLink.http.AppPool = appPool
+	httpConnLink.http.AppPool = strings.Replace(appPool, " ", "_", -1)
```
Comment (clarkb7) [inline pkg/network/etw/etw-http-service.go:754]: why the change in how the `appPool` is stored?

File: pkg/util/winutil/iisconfig.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		return err
+	}
+	idmap := make(map[uint32]string)
+
+	for _, site := range newcfg.ApplicationHost.Sites {
+		id, err := strconv.Atoi(site.SiteID)
+		if err != nil {
+			return err
+		}
+		idmap[uint32(id)] = strings.Replace(site.Name, " ", "_", -1)
```
Comment (clarkb7) [inline pkg/util/winutil/iisconfig.go:149]: why not store the site name as is?

Comment (derekwbrown) [reply pkg/util/winutil/iisconfig.go:149]: because the site name becomes a tag, and tags shouldn't hvae spaces.  And it's much more efficient to do it once when we parse it than to to the string convert every time we access it.

File: pkg/network/etw/etw-http-service.go (hunk 1 above)
Comment (derekwbrown) [reply pkg/network/etw/etw-http-service.go:754]: because it becomes a tag and it can't have spaces.

File: pkg/util/winutil/iisconfig.go (hunk 2 above)
Comment (clarkb7) [reply pkg/util/winutil/iisconfig.go:149]: Are the `SiteName` or `AppPool` variables used anywhere else that might expect the real value instead of a tag value?

If you must store them as tag values the variable name should reflect that (e.g. `SiteName_tag`)

Comment (derekwbrown) [reply pkg/util/winutil/iisconfig.go:149]: it is only used as a tag.
I can change the variable name.
The reason I don't want to make the transformation when the tag is made is that is potentially done thousands of times per second, vs. doing once when we read.

File: pkg/util/winutil/iisconfig.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/fsnotify/fsnotify"
+)
+
+var (
+	// make global so that we can override for tests.
+	iisCfgPath = filepath.Join(os.Getenv("windir"), "System32", "inetsrv", "config", "applicationHost.config")
```
Comment (richl-dd) [inline pkg/util/winutil/iisconfig.go:26]: Should we make this configurable in case a customer installed in a non-default location?

File: pkg/util/winutil/iisconfig.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+	var err error
+
+	iiscfg.watcher, err = fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	// check for existence
+	_, err = os.Stat(iisCfgPath)
+	if os.IsNotExist(err) {
```
Comment (richl-dd) [inline pkg/util/winutil/iisconfig.go:55]: Can we log an error here so it shows up in a flare and we can see that the path didn't exist at the expected location. Will help support cases since we know that is the reason.

**Review by richl-dd — COMMENTED** [review]
Aside from comment on configurable IIS config path, looks good to me.

File: pkg/network/etw/etw-http-service.go (hunk 1 above)
Comment (clarkb7) [reply pkg/network/etw/etw-http-service.go:754]: can you rename the `http.AppPool` and `http.SiteName` attributes as well, to indicate they contain tag names and not the actual application pool and site name.

File: pkg/util/winutil/iisconfig.go (hunk 4 above)
Comment (derekwbrown) [reply pkg/util/winutil/iisconfig.go:55]: since the error is returned, it is already logged by the caller.

File: pkg/util/winutil/iisconfig.go (hunk 3 above)
Comment (richl-dd) [reply pkg/util/winutil/iisconfig.go:26]: 👍

File: pkg/network/etw/etw-http-service.go (hunk 1 above)
Comment (iglendd) [reply pkg/network/etw/etw-http-service.go:754]: I actually do not like renaming functions or vars things here (in this module). They are converted to tags at different layer. Perhaps that where it should be replace spaces with underscores.

**Review by richl-dd — APPROVED** [review]

File: pkg/network/etw/etw-http-service.go [hunk 5]
```diff
@@ -152,15 +153,16 @@ type Http struct {
...
 
 	// <<<MORE ETW HttpService DETAILS>>>
 	// We can track FULL url and few other attributes. However it will require much memory.
 	// Search for <<<MORE ETW HttpService DETAILS>>> top find all places to be uncommented
 	// if such tracking is desired
 	//
 	// Url           string
-	SiteID uint32
+	SiteID   uint32
+	SiteName string
```
Comment (clarkb7) [inline pkg/network/etw/etw-http-service.go:165]: Can you rename this member `SiteNameAsTag` as well

**Review by clarkb7 — CHANGES_REQUESTED** [review]

**Review by clarkb7 — APPROVED** [review]
it might be worth it to add a comment to the struct explaining your reasoning for storing the tag names instead of the real names from the previous comment so someone doesn't refactor it later
> The reason I don't want to make the transformation when the tag is made is that is potentially done thousands of times per second, vs. doing once when we read.
